apps/web/src/utils/auth.ts

```diff
@@ -109,26 +109,37 @@ const providers: Provider[] = [
 
 // If we have an OAuth provider configured, we add it to the list of providers
 if (process.env.OIDC_NAME) {
+  let userinfo = {};
+  if (process.env.OIDC_FORCE_USER_INFO) {
+    userinfo = {
+      ...userinfo,
+      async request(context: any) {
```

Comment:
Next-auth docs suggest not using `userinfo` in this this way. 

https://next-auth.js.org/configuration/providers/oauth#userinfo-option

I think it's better to keep it simple and assume that the identity provider is openid connect compliant and will return the data we need in the id token. This means `idToken` should always be `true` and `userinfo` is not necessary.

It is much easier to add configuration down the line then to remove configuration.

Reply:
To be honnest, I am a bit perplexed with next-auth configuration for `userinfo`

`idToken` may not be true using an OAuth 2.0 instead of an openid connect provider. If `idToken` is false, `userinfo` will be called by NextAuth. This make sense.

---

> "The Claims requested by the profile, email, address, and phone scope values are returned from the UserInfo Endpoint, as described in [Section 5.3.2](https://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse), when a response_type value is used that results in an Access Token being issued. However, when no Access Token is issued (which is the case for the response_type value id_token), the resulting Claims are returned in the ID Token."

According to [openid specifications](https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims), if I understood them correctly, requested claims should be provided by `userinfo` endpoint if the `response_type` return an access_token. This is the case with `response_type=code` used by next-auth and most oidc clients.

In practice, I think there are clients using one or the other method independently of the specifications. I believe there would be benefits to supports getting claims from both `userinfo` and `id_token`.

Reply:
I could not make `OIDC_USER_INFO_URL` works with `OIDC_FORCE_USER_INFO` but I don't think this is really an issue as `OIDC_USER_INFO_URL` should be used for oauth 2.0 providers (that don't have a discovery_url) while `OIDC_FORCE_USER_INFO` are only useful for some oidc providers.